share/commands/directive-replace

```diff
@@ -28,15 +28,15 @@ shift  $((OPTIND-1))
 
 if [[ -z ${OLD_DIRECTIVE} || -z ${NEW_DIRECTIVE} ]]
 then
-  echo "Usage: rudder directive replace [-iIvdq] [-c] -o <old-directive> -n <new-directive>"
+  echo "Usage: rudder directive replace [-iIvd] [-c] -o <old-directive> -n <new-directive>"
   exit 1
 fi
 
 echo "DIRECTIVE ${OLD_DIRECTIVE} will be replaced by ${NEW_DIRECTIVE}"
 echo ""
 
 # Find the rules using the old directive
-RULES=$(complete_api_call "${API_URL}/api/latest/rules" "${TOKEN}" "GET" "| jq -r ' .data.rules[] | select(.directives[] | contains(\"${OLD_DIRECTIVE}\")) | {\"id\":.id, \"displayName\":.displayName, \"directives\":.directives}'" "${DISPLAY_COMMAND}")
+RULES=$(filtered_api_call "${API_URL}/api/latest/rules" "${TOKEN}" "GET" "| jq -r ' .data.rules[] | select(.directives[] | contains(\"${OLD_DIRECTIVE}\")) | {\"id\":.id, \"displayName\":.displayName, \"directives\":.directives}'" "${DISPLAY_COMMAND}")
```

Comment:
I'm still annoyed by the fact that you are evaluating a pipe in a variable in a sub function instead of simply using a pipe, this is not readable and error prone

Reply:
You never use $filter in the filtered_api_call function, so you don't need to pass it at all
Just remove the parameter

Reply:
it is used in the curl command crafting. And then executed with the eval

Reply:
No it is not used in the curl command and yes it is used in the eval since the goal is to remove the eval.
Calling the pipe directly is just a matter of removing quotes

Reply:
it is used to pass stuff to curl in the command 
"-d \"techniqueVersion=${TECHNIQUE_VERSION}\""

This could be avoided by passing it through the $url arg of filtered_api_call, but it would maybe make it harder to understand